src/Symfony/Component/TypeInfo/TypeResolver/StringTypeResolver.php

```diff
@@ -135,9 +135,14 @@ private function getTypeFromNode(TypeNode $node, ?TypeContext $typeContext): Typ
                 'bool', 'boolean' => Type::bool(),
                 'true' => Type::true(),
                 'false' => Type::false(),
-                'int', 'integer', 'positive-int', 'negative-int', 'non-positive-int', 'non-negative-int', 'non-zero-int' => Type::int(),
+                'int', 'integer' => Type::int(),
+                'positive-int' => Type::intRange(from: 1),
+                'negative-int' => Type::intRange(to: -1),
+                'non-positive-int' => Type::intRange(to: 0),
+                'non-negative-int' => Type::intRange(from: 0),
+                'non-zero-int' => Type::union(Type::intRange(to: -1), Type::intRange(from: 1)),
                 'float', 'double' => Type::float(),
-                'string',
+                'string' => Type::string(),
                 'class-string',
```

Comment:
I think we'll need to handle `class-string<Foo>`, and `interface-string<Bar>` in a specific way.
IMHO, we should split the work in 2 PRs, on for the integers stuff, and on other for the strings stuff.

Reply:
Yeah, I was wondering about that. Seems wrong to leave things half finished.
Happy to create a second PR. Not sure what the pattern is here - wait until this is merged, or base thge second off this branch?

Reply:
Sorry, I meant to move **all the string related stuff** (ie: `ExplicitString`) in another PR. By doing that, you won't have to base the second PR on the first.